tests/test_remove_styles.py

```diff
@@ -0,0 +1,290 @@
+        <activity id="3" modulename="quiz">
+            <quiz id="1">
+                <name>Fist Example Quiz</name>
+                <question_instances>
+                    <question_instance id="1">
+                        <questionid>1</questionid>
+                    </question_instance>
+                    <question_instance id ="2">
+                        <questionid>2</questionid>
+                    </question_instance>
+                </question_instances>
+            </quiz>
+            <quiz id="2">
+                <name>Second Example Quiz</name>
+                <question_instances>
+                    <question_instance id="3">
+                        <questionid>3</questionid>
+                    </question_instance>
+                </question_instances>
+            </quiz>
+        </activity>
+    """.strip()
+    quiz3_dir = tmp_path / "activities/quiz_3"
+    quiz3_dir.mkdir(parents=True)
+    (quiz3_dir / "quiz.xml").write_text(quiz3_content)
+
+    question_content = f"""
+    <?xml version="1.0" encoding="UTF-8"?>
+    <question_categories>
+        <question_category id="1">
+            <name>Quiz Bank 'Algebra1.1 Check Your Readiness'</name>
+            <questions>
+                <question id="1">
+                <questiontext>{html.escape(QUESTION1_CONTENT)}</questiontext>
+                <answers>
+                    <answer id="1">
+                    <answertext>{html.escape(ANSWER1_CONTENT)}</answertext>
+                    </answer>
+                    <answer id="2">
+                    <answertext>{html.escape(ANSWER2_CONTENT)}</answertext>
+                    </answer>
+                </answers>
+                </question>
+                <question id="2">
+                <questiontext>{html.escape(QUESTION2_CONTENT)}</questiontext>
+                </question>
+                <question id="3">
+                <questiontext>{html.escape(QUESTION3_CONTENT)}</questiontext>
+                </question>
+            </questions>
+        </question_category>
+    </question_categories>
+    """.strip()
+    (tmp_path / "questions.xml").write_text(question_content)
+
+    return tmp_path
+
+
+def test_remove_all_styles(mbz_path, tmp_path):
+    remove_styles.remove_styles(mbz_path)
+    with open(f"{tmp_path}/activities/lesson_1/lesson.xml", 'r') as f:
+        file = f.read()
+        assert ('style' not in file)
+    with open(f"{tmp_path}/activities/page_2/page.xml", 'r') as f:
+        file = f.read()
+        assert ('style' not in file)
+    with open(f"{tmp_path}/questions.xml", 'r') as f:
+        file = f.read()
+        assert ('style' not in file)
+
+
+def test_remove_styles_from_main(mbz_path, tmp_path, mocker):
+    mocker.patch(
+        "sys.argv",
+        ["", f"{tmp_path}"]
+    )
+    remove_styles.main()
+
+    with open(f"{tmp_path}/activities/lesson_1/lesson.xml", 'r') as f:
+        file = f.read()
+        assert ('style' not in file)
```

Comment:
1) Oh, right... good catch I did not notice that. I know @P-Gill97 ran into this issue on 91. Prabhdip, I'd be curious to see how you fixed this.  

2) I think I see what you mean here - that all the HTML sitting inside of a \<div\> element makes the IBs nested, i.e. not sequential / at the root of the tree. 

I'm confused why you mention that all the other methods would need to be updated to use etree_fragments as well?

Reply:
I think @P-Gill97 may have avoided this issue as he was completely rewriting the content to a single `<div></div>`.

The signature of `html.fromstring` and `html.fragments_fromstring` are different in that the latter returns a list. So where the methods currently expect to parse `self.etree` they'd need to do something like the `for fragment in self.etree_fragments` and iterate on all fragments. We probably need to include test content with fragments as well to minimize the risk of regressions (this would be another advantage of a shared fixture as we could have a single "copy" of the mbz used for relevant tests).

Reply:
In applying etree fragments to MoodleElements, I've broken something in the validator (it will throw a type error when parsing a full mbz) that doesn't break any unit tests. I'm still having trouble reproducing the error in new unit tests, but the bug is specific to this modification so I'm hesitant to create a new issue for it.

Reply:
@mjoss17 - Want to pair for a bit tomorrow after coffee break to see if extra :eyes: might help?

Reply:
@rnathuji that'd be great, I'd really appreciate👍

Reply:
Discovering more issues here as I test the `remove-styles` and `validate-mbz-html` commands with the new fragment architecture. Now the validator will create multiples of every error, anywhere from 8-16 copies of it for each moodle activtiy. Right now my guess is that MoodleElements are somehow creating multiple etree fragments when there should only be one, looking into it more, but something to note for this afternoon.